green/result.py

```diff
@@ -158,15 +161,15 @@ class BaseTestResult:
     I am inherited by ProtoTestResult and GreenTestResult.
     """
 
-    def __init__(self, stream, colors):
-        self.stdout_output = OrderedDict()
-        self.stderr_errput = OrderedDict()
+    def __init__(self, stream, *, colors: Colors | None = None):
```

Comment:
I'm not familiar with a lone `*` parameter. Does that mean accept and ignore any additional positional parameters?

Reply:
It is to state that the remainder must be passed as named arguments only. The TestResult class of unittest has this signature in 3.12:
```
def __init__(self, stream=None, descriptions=None, verbosity=None):
```

And I got a tad confused when reading the green codebase. Since we might want to mirror the upstream signature, it is safer to enforce `colors` to be a named parameter.

https://docs.python.org/3/reference/compound_stmts.html#function-definitions

> Parameters after “*” or “*identifier” are keyword-only parameters and may only be passed by keyword arguments. Parameters before “/” are positional-only parameters and may only be passed by positional arguments.

Reply:
👍🏻